### DISCUSSION TITLE
Searching for albums with length equal to or below 40 minutes & random searches

### DISCUSSION THREAD

**Discussion by zz0rk** [quodlibet/quodlibet#4330] [Q&A] [answered]
I am wondering if somebody could provide me with an example search how to find particular albums that meed a certain album length criteria, e.g. below or equal to 35 minutes; above 45 but below 65, etc. and how to combine with genre and year. So, search for a hard rock album that's 35 minutes long and came out between 1980 and 1985.

Secondly, there are some built in random filters Random Genre, Random Album, etc.
How would I create a search that provides let's say 5 random metal albums that came out in 2020.

Thank you very much.

**Comment by zz0rk** [comment]
I just want to clarify that I know how to search by year range and genre. Just not the random and album length bit.
&(genre = power metal, #(2018 <= year <= 2020))
&(genre = power metal, #(year = 2023))
Thank you

**Comment by bassstorm** [comment]
Hi!

I preferred to have queries in small chunks, which could be combined later (you need to enable a respective plugin).

```
release
#(length > 16:00)

good tracks
#(rating >= 0.8)

top albums
&(#(rating:avg >= 0.7), @(saved: release))

jungle fever
&(genre=jungle, @(saved: good tracks))
```

Above allows you to work with simple concepts, then wiring them up with boolean operands according actual needs. Hope you get the idea.

P.S. Have no exact answer for Random part. And no QL installed here to check. Maybe there was some plugin for random albums, but I didn't have such a need.

**Comment by zz0rk** [comment, marked as answer]
Hiya, thanks for your reply.
I found answers to some of my questions myself. I am posting them here. Maybe they will be helpful to others. Please feel free to close this ticket and thank you for your work on this project.

#(rating = 1)
Rating One

&(genre = genre, #(year = year), #(rating = 1))
Rating One & Genre & Year

&(artist = artist, #(length <= 50:00))
Artist & Album Length

&(genre = genre, #(year = year))
Genre & Year

&(genre = genre, #(year = year), #(length <= 50:00))
Genre & Year & Album Length

&(genre = genre, #(yearStart <= year <= yearEnd))
Genre & Year Range

&(genre = genre, #(yearStart <= year <= yearEnd), #(length <= 50:00))
Genre & Year Range & Album Length
